k2/csrc/ragged.cu

```diff
@@ -291,6 +291,26 @@ int32_t RaggedShape::operator[](const std::vector<int32_t> &indexes) {
   return cur_idx;
 }
 
+int32_t RaggedShape::TotSize(int32_t axis) const {
+  K2_CHECK_GE(axis, 0);
+  K2_CHECK_LT(axis, NumAxes());
+  if (axis == 0)
+    return Dim0();
+  else {
+    const RaggedShapeDim &rsd = axes_[axis - 1];
+    if (rsd.cached_tot_size >= 0) {
+      return rsd.cached_tot_size;
+    } else {
+      // if we had row_ids set up, we should have set cached_tot_size.
+      K2_CHECK_EQ(rsd.row_ids.Dim(), 0);
+      K2_CHECK_GT(rsd.row_splits.Dim(), 0);
+      const_cast<RaggedShapeDim &>(rsd).cached_tot_size =
```

Comment:
I would recommed declaring `cached_tot_size` as mutable:

```cpp
mutable int32_t cached_tot_size;
```
so that const cast can be eliminated.

Reply:
Actually `const_cast` is only used here for `cached_tot_size`, no other code does this. So I prefer to not change to `mutable`